### PR TITLE
Support relative imports in AnsiballZ.

### PR DESCRIPTION
**Pull request by mattclay**
##### SUMMARY

The module_common changes are a complete hack intended only to get the new integration tests passing.

The tests demonstrate the kinds of relative imports that should be supported.
The final changes to implement this feature should be able to pass the tests.

This is a proof-of-concept implementation for https://github.com/ansible/ansible/issues/59465

##### ISSUE TYPE

Feature Pull Request

##### COMPONENT NAME

lib/ansible/executor/module_common.py

### CODE REVIEW

**Comment by mattclay** [comment]
No attempt was made to make the `lib/ansible/executor/module_common.py` changes even close to what they would need to be for production use. The goal of this PR was to demonstrate passing tests with the desired features in use.

File: docs/docsite/rst/porting_guides/porting_guide_2.9.rst [hunk 1]
```diff
@@ -40,6 +40,38 @@ Modules
...
+  .. code-block:: python
+
+    # File: ansible_collections/my_namespace/my_collection/plugins/modules/my_module.py
+    # Old way to use an absolute import to import module_utils from the collection:
+    from ansible_collections.my_namespace.my_collection.plugins.module_utils import my_util
+    # New way using a relative import:
+    from ..module_utils import my_util
+
+  Modules and module_utils shipped with Ansible can use relative imports as well but the savings
+  aren't as great:
```
Comment (acozine) [inline docs/docsite/rst/porting_guides/porting_guide_2.9.rst]:
```suggestion
  are smaller:
```

File: docs/docsite/rst/porting_guides/porting_guide_2.9.rst [hunk 2]
```diff
@@ -40,6 +40,38 @@ Modules
...
+
+  .. code-block:: python
+
+    # File: ansible/modules/system/ping.py
+    # Old way to use an absolute import to import module_utils from core:
+    from ansible.module_utils.basic import AnsibleModule
+    # New way using a relative import:
+    from ...module_utils.basic import AnsibleModule
+
+  The number of dots (``.``) to use depend on the depth in the tree that the file is located.
```
Comment (acozine) [inline docs/docsite/rst/porting_guides/porting_guide_2.9.rst]:
```suggestion
  Each single dot (``.``) represents one level of the tree (equivalent to ``../`` in filesystem relative links).
```

**Review by acozine — COMMENTED** [review]
Overall looks good. This is a nice feature.

I made two small suggestions for the docs portion.

**Review by AlanCoding — APPROVED** [review]

**Pull request merged**